File: backend/src/main/java/io/github/xpax/syllabi/entity/Teacher.java
package io.github.xpax.syllabi.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Teacher {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;
    private String surname;
    private String title;
    private String phone;
    private String email;
    private String pbnId;

    @JsonIgnore
    @OneToOne(cascade = {CascadeType.PERSIST})
    @JoinColumn(name = "teacher_id")
    private Job job;

    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/src/app/entity/teacher.ts
import { Job } from "./job";
import { UserWithOnlyId } from "./user-with-only-id";

export interface Teacher {
    id: number;
    name: string;
    surname: string;
    title: string;
    phone: string;
    email: string;
    pbnId: string;
    user: UserWithOnlyId;
    job: Job;
}